### PR TITLE
Lifetime vis v1.0.0

### PR DESCRIPTION
**Pull request by Alaric617R**
1. Added all new style (css template) to support lifetime visualization. Compatible with original rustviz
2. Added new module, lifetime visualization, in SVG front end.
3. Modified SVG generation logic, now supports lifetime visualization.
4. Fixed bug in original rustviz that code panel width is once unspecified.
5. Added examples using lifetime visualization, in src/examples
6. Created manual of using lifetime visualization. Proofreading still needed.

### CODE REVIEW

File: rustviz_mdbook/view_examples.sh [hunk 1]
```diff
@@ -24,33 +24,37 @@ printf "Generating visualizations for the following examples: \n"
...
-    "nll_lexical_scope_different"
-    "printing"
-    "string_from_move_print"
-    "string_from_print"
-    "struct_lifetime"
-    "struct_rect"
-    "struct_rect2"
-    "struct_string"
-    "extra_credit"
+    # "copy"
```
Comment (cyrus-) [inline rustviz_mdbook/view_examples.sh]: why were all these examples deleted / commented out? don't want that in the base branch...

File: src/examples/copy/main.rs [hunk 2]
```diff
@@ -1,16 +1,11 @@
 /* --- BEGIN Variable Definitions ---
-Owner x; Owner y
+Owner x; Owner y; Owner k;
```
Comment (cyrus-) [inline src/examples/copy/main.rs]: this should be a new example, not a change to the copy example?

File: test_log.md [hunk 3]
```diff
@@ -1,3 +1,7 @@
+## Testing
```
Comment (cyrus-) [inline test_log.md]: what is this file?

**Review by cyrus- — CHANGES_REQUESTED** [review]
- not sure why a bunch of examples were deleted (see comment)
- some .DS_Store files mistakenly got included
- see a few other comments
(this is just a quick partial review, will go through and try to install and play with everything once this stuff is fixed)

File: test_log.md (hunk 3 above)
Comment (Alaric617R) [reply test_log.md]: will delete this in new pr.

File: src/examples/copy/main.rs (hunk 2 above)
Comment (Alaric617R) [reply src/examples/copy/main.rs]: yeah, that seems to be when i want to test some basic functionality of lifetime visualization using the existing framework of rustviz. Will unroll that change.

File: rustviz_mdbook/view_examples.sh (hunk 1 above)
Comment (Alaric617R) [reply rustviz_mdbook/view_examples.sh]: will uncomment those. I commented those for the convenience of testing all lifetime vis funtionalities.

**Pull request closed without merging**